### PR TITLE
Refactor `AxPropertyDescriptor` to replace `ArrayList`

### PR DESCRIPTION
**Pull request by elachlan**
Refactored `AxPropertyDescriptor` to replace `ArrayList` with `List<T>`

Related: #8140

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8152)

### CODE REVIEW

**Comment by elachlan** [comment]
There is a section where there is a `CopyTo` call:
https://github.com/dotnet/winforms/blob/70a0680cb90e48502a2a119c4df6be1649c41f90/src/System.Windows.Forms/src/System/Windows/Forms/AxHost.AxPropertyDescriptor.cs#L276-L278

Should I use ToArray() on the List instead? I wanted to keep the changes to a minimum hence why I didn't do it.

File: src/System.Windows.Forms/src/System/Windows/Forms/AxHost.AxPropertyDescriptor.cs [hunk 1]
```diff
@@ -268,7 +267,7 @@ internal void UpdateAttributes()
                     return;
                 }
 
-                ArrayList attributes = new ArrayList(AttributeArray);
+                List<Attribute> attributes = new List<Attribute>(AttributeArray);
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/AxHost.AxPropertyDescriptor.cs:270]:
```suggestion
                List<Attribute> attributes = new(AttributeArray);
```

**Comment by dreddy-work** [comment]
> Should I use ToArray() on the List instead? I wanted to keep the changes to a minimum hence why I didn't do it.

I don't see why not. it creates a new copy of array.

**Review by RussKie — APPROVED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/AxHost.AxPropertyDescriptor.cs [hunk 2]
```diff
@@ -268,16 +267,13 @@ internal void UpdateAttributes()
                     return;
                 }
 
-                ArrayList attributes = new ArrayList(AttributeArray);
+                List<Attribute> attributes = new(AttributeArray);
                 foreach (Attribute attr in _updateAttributes)
                 {
                     attributes.Add(attr);
```
Comment (KindDragon) [inline src/System.Windows.Forms/src/System/Windows/Forms/AxHost.AxPropertyDescriptor.cs:273]: I think this foreach equal to 
attributes.AddRange(_updateAttributes);

Comment (elachlan) [reply src/System.Windows.Forms/src/System/Windows/Forms/AxHost.AxPropertyDescriptor.cs:273]: Good idea. Could you put in a PR?

**Pull request merged**